Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Feto;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using TMPro;
+
+public class GameManager : Singleton<GameManager>
+{
+    [SerializeField] GameManagerScriptable _settings;
+    private int iterations;
+    GameStates state;
+
+    private void Start() {
+        iterations = 0;
+        UpdateGameState(GameStates.Play);
+    }
+
+    public void UpdateGameState(GameStates newState) {
+        state = newState;
+        switch (state) {
+            case GameStates.Play:
+                if(iterations < _settings.totalIterations) {
+                    StartCoroutine(Playing());
+                } else {
+                    UpdateGameState(GameStates.GameOver);

```

Comment:
Shoultdn't that in inside Crier, since it's the state called when finishing hte playing coroutine?